### PR TITLE
Add CSRF-protection

### PR DESCRIPTION
**Pull request by tavriaforever**
// cc @tadatuta @tormozz48 @gela-d

### CODE REVIEW

File: desktop.blocks/comments/_view/comments_view_close.bemtree [hunk 1]
```diff
@@ -1,8 +1,7 @@
 block('comments').mod('view', 'close')(
     content()(function() {
         var comments = this.ctx.data,
```
Comment (tadatuta) [inline desktop.blocks/comments/_view/comments_view_close.bemtree]: `var ctx = this.ctx`

File: desktop.blocks/comments/__add-form/comments__add-form.bemtree [hunk 2]
```diff
@@ -33,6 +33,13 @@ block('comments').elem('add-form')(
                         ]
                     },
                     {
+                        block: 'input',
```
Comment (tadatuta) [inline desktop.blocks/comments/__add-form/comments__add-form.bemtree]: just use `{ tag: 'input', attrs: { type: 'hidden', name: '_csrf', val: req.csrf } }` instead.
input block is too huge for this.

File: desktop.blocks/issue/issue.bemtree [hunk 3]
```diff
@@ -262,6 +262,13 @@ block('issue')(
                     },
                     {
                         block: 'input',
+                        mods: { type: 'hidden' },
```
Comment (tadatuta) [inline desktop.blocks/issue/issue.bemtree]: same as above

File: desktop.blocks/input/_type/input_type_hidden.bemhtml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+block('input').mod('type', 'hidden').elem('control').attrs()(function() {
```
Comment (tadatuta) [inline desktop.blocks/input/_type/input_type_hidden.bemhtml]: you won't need this template after switching to `{ tag: 'input', /* ... */ }`

File: desktop.blocks/issue/issue.js [hunk 5]
```diff
@@ -75,17 +75,20 @@ modules.define('issue', ['i-bem__dom', 'jquery', 'events__channels'], function(p
...
-                var _this = this;
-
                 $.ajax({
                     dataType: 'html',
                     type: 'PUT',
-                    data: [{ name: 'state', value: 'closed' }, { name: 'number', value: this.params.number }],
-                    url: this.params.forumUrl + 'issues/' + this.params.id + '/?__mode=json'
+                    data: {
+                        state: 'closed',
+                        number: this.params.number,
```
Comment (tadatuta) [inline desktop.blocks/issue/issue.js]: you may cache `this.params` into variable

**Comment by tavriaforever** [comment]
up

File: desktop.blocks/input/_type/input_type_hidden.bemhtml (hunk 4 above)
Comment (tadatuta) [inline desktop.blocks/input/_type/input_type_hidden.bemhtml]: why have you kept it?

Comment (tavriaforever) [inline desktop.blocks/input/_type/input_type_hidden.bemhtml]: It`s mistake - i will remove it

**Comment by tavriaforever** [comment]
up2

**Comment by tadatuta** [comment]
lgtm

**Pull request merged**